src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,891 @@
+        self._shutdown_flag.set()
+        self._listener_thread.join()  # wait till the listener thread closes
+        self._socket.shutdown(socket.SHUT_RDWR)
+        self._socket.close()
+
+    def isMeasuring(self):
+        '''
+        Returns true if an acquisition is running. False otherwise.
+        '''
+        return self._measuring_flag.is_set()
+
+    def waitTillMeasurementStarts(self):
+        self._measuring_flag.wait_true()
+
+    def waitTillMeasurementComplete(self):
+        logging.debug("Waiting for measurmenet to complete...")
+        self._measuring_flag.wait_false()
+        logging.debug("Measurement complete. Done waiting. ")
+
+    def _sendMessage(self, msg):
+        '''
+        Sends an order to the device.
+        msg: Message type object
+        '''
+        logging.debug("S: Msgtype %s", msg)
+        time.sleep(0.02)  # wait a bit to prevent flooding the server.
+        self._net_access.acquire()
+        self._socket.sendall(msg.to_bytes())
+        self._net_access.release()
+
+    def StartMeasurement(self):
+        '''
+        Starts the measurement process with the values stored in the object config
+        '''
+        logging.info("Requesting an acquisition. ")
+
+        # determine measurement parameters
+        pixel_size = self._metadata[model.MD_PIXEL_SIZE][0]
+        if self._metadata[model.MD_PIXEL_SIZE][0] != self._metadata[model.MD_PIXEL_SIZE][1]:
+            logging.warning("Pixel size %s mismatch. Should be symmetrical. ", self._metadata[model.MD_PIXEL_SIZE])
+        iScanningPattern = int(self.scanner.bidirectional.value)
+        iPixelNumber_X, iPixelNumber_Y = self.scanner.resolution.value
+
+        # Define optional data records to send to the server
+        optional_data = []
+        optional_data.append(OptionalDataRecord("TimePerPixel", PQ_OPT_DATATYPE_FLOAT,
+                                        self.scanner.dwellTime.value))
+
+        # Only send a filename if it is not empty. Otherwise we will have problems.
+        # If no filename is specified, the default filename from the server will be adopted in the VA
+        if self.scanner.filename.value != "":
+            optional_data.append(OptionalDataRecord("Filename", PQ_OPT_DATATYPE_FIXED_LENGTH_STRING,
+                                        os.path.splitext(self.scanner.filename.value)[0]))
+        else:
+            logging.warning("No filename specified. Symphotime controller will specify filename.")
+
+        if self.scanner.directory.value != "":
+            optional_data.append(OptionalDataRecord("Groupname", PQ_OPT_DATATYPE_FIXED_LENGTH_STRING,
+                                        self.scanner.directory.value))
+        else:
+            logging.warning("No directory name specified. Symphotime controller will specify groupname.")
+
+        optional_data.append(OptionalDataRecord("Objective", PQ_OPT_DATATYPE_FIXED_LENGTH_STRING,
```

Comment:
Isn't there a field called like "user notes" or "experiment description"? It would fit better to MD_DESCRIPTION. For the "Objective" field, I'd suggest to link it to the `MD_LENS_NAME` metadata. Also, only add this optional record if the metadata is present.